docs: expand community docs: contributor - simple contribution flow a…

ezinneanne opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

- docs: expand community docs: contributor - simple contribution flow 


**Related issue(s)**
<!-- If you refer to a particular issue, provide its number, otherwise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->
#1622 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Documentation**
	- Introduced a new guide titled "Simple contribution flow for contributors," outlining the contribution process, prerequisites, and effective steps for contributors. The guide includes a checklist to help ensure quality and clarity in contributions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

thulieblack commented:
@ezinneanne separate the files into 2 PRs

ezinneanne commented:
@thulieblack @bandantonio I have separated the PR files and also updated this contribution document by adding contribution steps and checklist. Kindly review

thulieblack reviewed (commented):
Great start Ezinne:

- your file should be under the onboarding folder and front matter
- rename file to `contribution-flow.md` for consistency with other files
- detail the whole process for the contribution flow based from the https://github.com/asyncapi/website/blob/master/CONTRIBUTING.md#contribution-flow.  in your doc. 

Love the checklist

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+# Contributor - Simple Contribution Flow
+This article explains how to contribute effectively to AsyncAPI. It guides you through the contribution process and provides a checklist to ensure high-quality contributions. 
```
```suggestion
This guide explains how to contribute effectively to AsyncAPI. It guides you through the contribution process and provides a checklist to ensure high-quality contributions. 
```

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+
+
+## Making a Contribution
+We welcome your contributions to AsyncAPI! Whether you have suggestions, improvements, or new ideas, we encourage you to participate.
```
```suggestion
We welcome your contributions to AsyncAPI, whether you have suggestions, improvements, or new ideas.
```

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+- Read the [onboarding contributor guide](https://www.asyncapi.com/docs/community/onboarding-guide).
+- If you are contributing to documentation, review the [documentation onboarding checklist](https://www.asyncapi.com/docs/community/onboarding-guide/docs-onboarding-checklist). 
+- For other repositories, refer to the [general contribution guidelines](https://github.com/asyncapi/community/blob/master/CONTRIBUTING.md).
+- Look through existing issues and pull requests to gain a better understanding of the repository.
```
```suggestion
- Look through existing issues and pull requests to understand the repository better.
```

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+- Search for [good first issues](https://github.com/issues?page=1&q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+label%3A%22good+first+issue%22). 
+- When you do find one, request to be assigned.
+- However, if you notice a bug, typo, or improvement, feel free to create an issue.
+- Make a PR only when you have been assigned by a maintainer.
```
```suggestion
- Make a PR only when a maintainer has assigned you.
```

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+- When you do find one, request to be assigned.
+- However, if you notice a bug, typo, or improvement, feel free to create an issue.
+- Make a PR only when you have been assigned by a maintainer.
+- Even if you created the issue yourself, wait to be assigned by a maintainer before you make a PR.
```
```suggestion
- Even if you created the issue yourself, wait for it to be assigned by a maintainer before you make a PR.
```

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+- However, if you notice a bug, typo, or improvement, feel free to create an issue.
+- Make a PR only when you have been assigned by a maintainer.
+- Even if you created the issue yourself, wait to be assigned by a maintainer before you make a PR.
+- Follow the [conventional commits guide](https://github.com/asyncapi/community/pull/1733) to write a good PR title.
```
Here, we need to use a relative URL instead of the PR link

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+- Make a PR only when you have been assigned by a maintainer.
+- Even if you created the issue yourself, wait to be assigned by a maintainer before you make a PR.
+- Follow the [conventional commits guide](https://github.com/asyncapi/community/pull/1733) to write a good PR title.
+- Remember to follow the [git workflow document](https://github.com/asyncapi/community/blob/master/git-workflow.md) when making a PR.
```
Same applies to the above. Use a relative URL

thulieblack commented on making-a-contribution.md:
```diff
@@ -0,0 +1,48 @@
+- Remember to follow the [git workflow document](https://github.com/asyncapi/community/blob/master/git-workflow.md) when making a PR.
+- After making a PR, patiently wait for reviews.
+- Make sure all your PR checks pass (turn green) to speed up the merging process.  
+- If your PR hasn’t been reviewed after a certain period, you can gently remind a maintainer on Slack or GitHub. Delays can happen for various reasons, but no one is ignoring you! 🙏🏼 .
```
explain they have to wait 1 1/2 weeks before pinging the maintainer on the PR and followup on Slack as last resort

ezinneanne commented:
@thulieblack I have made the changes but the relative URLs are a bit confusing as the links become inactive when I use a relative path. is there a way I could use relativ URLs and the links would still work?

thulieblack commented:
They'll work once we have them on the website

thulieblack commented:
don't forget to add the 3rd point to your doc https://github.com/asyncapi/community/pull/1730#pullrequestreview-2681295398

ezinneanne commented:
@thulieblack I have updated the specified URLs and added the 3rd point as well. I wanted to confirm if those URLs are the only ones that am to modify as relative URLs or all of the links in the doc.

thulieblack commented:
Thanks @ezinneanne for now,let'ss leave as is until #1833 is done and merged

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -0,0 +1,82 @@
+- Make a PR only when a maintainer has assigned you.
+- Even if you created the issue yourself, wait for it to be assigned by a maintainer before you make a PR.
+- Follow the [conventional commits guide](/asyncapi/community/pull/1733) to write a good PR title.
+- Remember to follow the [git workflow document](/asyncapi/community/blob/master/git-workflow.md) when making a PR.
```
```suggestion
- Follow the [git workflow document](/asyncapi/community/blob/master/git-workflow.md) when working on your changes and making a PR.
```

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -0,0 +1,82 @@
+- However, if you notice a bug, typo, or improvement, feel free to create an issue.
+- Make a PR only when a maintainer has assigned you.
+- Even if you created the issue yourself, wait for it to be assigned by a maintainer before you make a PR.
+- Follow the [conventional commits guide](/asyncapi/community/pull/1733) to write a good PR title.
```
Never link to documents that haven't been merged yet. If you have to, add a comment (or any other reminder) to add a link later, once a specific document is merged to master.

```suggestion
- Follow the conventional commits guide to efficiently describe your changes and maintain clear git history.
```

ezinneanne commented on docs/onboarding-guide/contribution-flow.md:
````diff
@@ -0,0 +1,82 @@
+## Contribution Flow
+This is a summary of the ideal contribution flow.
+
+```
+    ┌───────────────────────┐
+    │                       │
+    │    Open an issue      │
+    │  (a bug report or a   │
+    │   feature request)    │
+    │                       │
+    └───────────────────────┘
+               ⇩
+    ┌───────────────────────┐
+    │                       │
+    │  Open a Pull Request  │
+    │   (only after issue   │
+    │     is approved)      │
+    │                       │
+    └───────────────────────┘
+               ⇩
+    ┌───────────────────────┐
+    │                       │
+    │   Your changes will   │
+    │     be merged and     │
+    │ published on the next │
+    │        release        │
+    │                       │
+    └───────────────────────┘
+```
````
I was thinking a visual representation would be ideal for beginners to understand. Moreover, using a numbered list would mean repetition/redundancy of previous points mentioned above or should I remove it completely?

thulieblack replied:
We should have the diagram, but try using mermaid.js @ezinneanne because that's the one we use for docs and for consistency. https://mermaid.js.org/

ezinneanne commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -0,0 +1,82 @@
+### Before You Start Contributing
+- If you are joining AsyncAPI through [Slack](https://asyncapi.com/slack-invite) or an event, take your time to explore the [AsyncAPI website](https://www.asyncapi.com/).  
+- Read our [Code of Conduct](https://github.com/asyncapi/.github/blob/master/CODE_OF_CONDUCT.md).  
+- [Attend meetings](https://www.asyncapi.com/community/events) and engage in discussions in the [Slack channel](https://asyncapi.com/slack-invite).
```
Which channel should I mention? Although, I added the slack link because I noticed it was a common pattern in past onboarding content.

bandantonio replied:
TBH, I don't know, it was my question to clarify. Because, if I were a new community member, I would like to know what channels to check in the first place.
I'm OK to leave the invite link but from my perspective it looks a bit confusing. @thulieblack what do you think?

ezinneanne commented:
@bandantonio I have made the requested changes. I await your feedback and review.

bandantonio commented:
> Should I leave the #1733 link as it is or remove it like you suggested earlier.

@ezinneanne Ideally, you can do a workaround by hardcoding a link to the document beforehand, like this:

```md
1. Follow the [conventional commits guide](/asyncapi/community/blob/master/docs/onboarding-guide/conventional-commits.md) to efficiently describe your changes and maintain clear git history.
```

ezinneanne commented:
@bandantonio I have hardcoded the link. I believe it will run when it is merged. The only changes pending now are the ones you asked @thulieblack for her inputs.

thulieblack commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -0,0 +1,82 @@
+### Steps to making a good contribution
+- Once you are ready to contribute, set up your workflow environment using the [git workflow guide](/asyncapi/community/blob/master/git-workflow.md).
+- Search for [good first issues](https://github.com/issues?page=1&q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+label%3A%22good+first+issue%22). 
+- When you do find one, request to be assigned.
```
Oh no, don't mention the _'can you assign me_' statement. We are implementing a bot to flag that as spam.

thulieblack replied:
@bandantonio @ezinneanne any suggestions here

ezinneanne replied:
I'll agree with @bandantonio

thulieblack replied:
Okay then, over to you @ezinne

thulieblack commented:
For the links @ezinneanne use the format similar to the checklist https://github.com/asyncapi/community/edit/master/docs/onboarding-guide/docs-onboarding-checklist.md

It'll render once we have it on the doc merged on the website

ezinneanne commented:
@thulieblack @bandantonio I have made the changes, updated the links and used mermaid.js. Kindly review

thulieblack commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -0,0 +1,61 @@
+---
+title: Simple contribution flow for contributors
+weight: 130
+---
```
this is the correct path

thulieblack commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -0,0 +1,82 @@
+### Before Making a Contribution
+- Read the [onboarding contributor guide](https://www.asyncapi.com/docs/community/onboarding-guide).
+- If you are contributing to documentation, review the [documentation onboarding checklist](https://www.asyncapi.com/docs/community/onboarding-guide/docs-onboarding-checklist). 
+- For other repositories, refer to the [general contribution guidelines](https://github.com/asyncapi/community/blob/master/CONTRIBUTING.md).
```
@ezinneanne check this review

bandantonio replied:
This comment is no longer relevant, @ezinneanne updated the previous wordings and this phrase looks ok now.

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
````diff
@@ -0,0 +1,61 @@
+- Make sure all your PR checks pass (turn green) to speed up the merging process.  
+- If your PR hasn’t been reviewed after a week and half, you can gently remind a maintainer on Slack or GitHub. Delays can happen for various reasons, but no one is ignoring you! 🙏🏼 .
+
+## Contribution Flow
+This is a summary of the ideal contribution flow.
+
+```mermaid
+flowchart TD
+    A[Open an issue - a bug report or a feature request]
+    A --> B[Open a Pull Request - only after issue is approved]
+    B --> C[Your changes will be merged and published on the next release]
+```
+
````
```suggestion
```

thulieblack commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -0,0 +1,84 @@
+4. If you noticed a bug, typo, or a needed improvement, feel free to create an issue.
+5. **Make a PR only after a maintainer assigned the issue to you** even if you created the issue yourself.
+6. Follow the [conventional commits guide](conventional-commits) to write a good PR title.
+7. Follow the [git workflow document](/asyncapi/community/blob/master/git-workflow.md) when working on your changes and making a PR.
```
```suggestion
7. Follow the [git workflow document](conventional-commits) when working on your changes and making a PR.
```

thulieblack replied:
It will open once merged so don't worry

bandantonio replied:
@thulieblack Hmm, the link to the git workflow document was correct, could you please elaborate on this suggestion?

thulieblack replied:
The link shouldn't take them to the folder in GitHub but to the doc on the website. For example, how we did with the checklist https://github.com/asyncapi/community/blob/master/docs/onboarding-guide/docs-onboarding-checklist.md?plain=1

bandantonio replied:
Ok, I got it. But this link was referring to the git workflow document that isn't listed on the docs website, that's why it was a direct link to GitHub markdown. Conventional commits is a different document, I believe we will confuse readers with such a substitution.

thulieblack replied:
Then should we bring git workflow under the onboarding docs then? @bandantonio

ezinneanne replied:
@thulieblack you may be using wrong link here. It should be git-workflow instead of conventional-commits

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -25,26 +25,49 @@ At AsyncAPI, we value **quality over quantity**. This guide will help you make m
+
+### ✅ What You Should Always Do
+
+- **Be patient**. Maintainers are volunteers with jobs, families, and responsibilities. It may take time for your pull request (PR) or issue to be reviewed.
```
Full wording doesn't make sense here and brings inconsistency as you've already used the abbreviation earlier.
 
```suggestion
- **Be patient**. Maintainers are volunteers with their jobs, families, and responsibilities. It may take time for your PR or issue to be reviewed.
```

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -25,26 +25,49 @@ At AsyncAPI, we value **quality over quantity**. This guide will help you make m
+- **Be patient**. Maintainers are volunteers with jobs, families, and responsibilities. It may take time for your pull request (PR) or issue to be reviewed.
+- **Use the right Slack channels**. If you’re unsure where to post, ask politely or take a moment to check.
+- **Keep Slack tidy**. Use threads for conversations, avoid tagging `@maintainers` unless truly necessary, and don’t spam.
+- **Follow our [Slack etiquette](https://github.com/asyncapi/community/blob/master/slack-etiquette.md)** **and [Code of Conduct](https://github.com/asyncapi/.github/blob/master/CODE_OF_CONDUCT.md)**.
```
```suggestion
- **Follow our [Slack etiquette](/asyncapi/community/blob/master/slack-etiquette.md)** **and [Code of Conduct](/asyncapi/.github/blob/master/CODE_OF_CONDUCT.md)**.
```

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -25,26 +25,49 @@ At AsyncAPI, we value **quality over quantity**. This guide will help you make m
+
+### ❌ What You Should Never Do
+
+- Don’t tag maintainers repeatedly or demand attention.
+- Don’t post the same message in multiple places.
+- Don’t expect instant replies.
+- Don’t ignore the onboarding materials and contribution guidelines.
```
I would place the last item first as it is vitally important one. Basically when you ignore onboarding materials, you will fail with the rest of the mentionned items.

```suggestion
- Don't ignore the onboarding materials and contribution guidelines.
- Don't tag maintainers repeatedly or demand attention.
- Don't post the same message in multiple places.
- Don't expect instant replies.
```

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -25,26 +25,49 @@ At AsyncAPI, we value **quality over quantity**. This guide will help you make m
+- Repeated behavior could lead to permanent removal from the community.
+- PRs or issues may be closed or rejected if the behavior affects maintainers or the community.
+
+These rules are not meant to scare anyone—they’re here to make sure everyone can enjoy a healthy and respectful space. When in doubt, be thoughtful, ask questions politely, and refer to our onboarding and etiquette docs.
```
```suggestion
These rules are not meant to scare anyone - they’re here to make sure everyone can enjoy a healthy and respectful space. When in doubt, be thoughtful, ask questions politely, and refer to our onboarding and etiquette docs.
```

bandantonio reviewed (changes requested):
Just a few minor improvements

ezinneanne commented:
@thulieblack @bandantonio i have made the changes

thulieblack reviewed (dismissed):
Great work!

bandantonio commented on docs/onboarding-guide/contribution-flow.md:
```diff
@@ -28,33 +28,32 @@ At AsyncAPI, we value **quality over quantity**. This guide will help you make m
-- Don’t post the same message in multiple places.
-- Don’t expect instant replies.
-- Don’t ignore the onboarding materials and contribution guidelines.
+- Ignore essential onboarding materials and contribution guidelines.
+- Tagging maintainers repeatedly or demanding attention.
+- Posting the same message in multiple places.
+- Expecting instant replies.
```
Unfortunately, this update brought an inconsistency to the list. Thank about the statement: "... you should never **do**" not "you should never be doing", so your items show follow the same pattern: ignore, tag, post, expect. 
For example, check your previous list (lines 46-49) that follows this pattern: be, use, keep, follow.

ezinneanne commented:
@bandantonio i have made the update.

thulieblack commented:
/rtm
